cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -416,7 +415,9 @@ public ApplicationMeta getLatest(ApplicationReference appReference) throws IOExc
         appReference.app(ApplicationId.DEFAULT_VERSION));
     ApplicationMeta appMeta = appSpecTable.read(fields).map(this::decodeRow).orElse(null);
     if (appMeta != null) {
-      return appMeta;
+      // explicitly set the isLatest field to true
+      return new ApplicationMeta(
+          appMeta.getId(), appMeta.getSpec(), appMeta.getChange(), appMeta.getSourceControlMeta(), true);
```

Comment:
we should be passing true to the decodeRow() method instead of re-creating the object here. Otherwise it will be easy to misuse decodeRow() in other parts of the code.